pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/AggregationFunctionFactory.java

```diff
@@ -101,7 +101,8 @@ public static AggregationFunction getAggregationFunction(FunctionContext functio
         } else if (numArguments == 2) {
           // Double arguments percentile (e.g. percentile(foo, 99), percentileTDigest(bar, 95), etc.) where the
           // second argument is a decimal number from 0.0 to 100.0.
-          double percentile = parsePercentileToDouble(arguments.get(1).getLiteral());
+          // Have to use literal string because we need to cast int to double here.
+          double percentile = parsePercentileToDouble(arguments.get(1).getLiteralString());
```

Comment:
here and the rest of the place. can we do implicit casting using PinotDataType? instead of using Stirng?

Reply:
implicit cast doesn't work since we can cast integer to double. We should do explicit cast.
but this is not in this PR's scope. I can do this in next PR. Want to keep this PR's scope small for easier management.